src/NUnitFramework/framework/Constraints/RangeConstraint.cs

```diff
@@ -33,26 +33,35 @@ namespace NUnit.Framework.Constraints
     /// </summary>
     public class RangeConstraint : Constraint
     {
-        private readonly IComparable from;
-        private readonly IComparable to;
+        private readonly object from;
+        private readonly object to;
 
         private ComparisonAdapter comparer = ComparisonAdapter.Default;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="RangeConstraint"/> class.
         /// </summary>
-        /// <remarks>from must be less than or equal to true</remarks> 
-        /// <param name="from">Inclusive beginning of the range. Must be less than or equal to to.</param>
-        /// <param name="to">Inclusive end of the range. Must be greater than or equal to from.</param>
+        /// <param name="from">Must be less than or equal to the <paramref name="to"/> value.</param>
+        /// <param name="to">Must be greater than or equal to the <paramref name="from"/> value.</param>
         public RangeConstraint(IComparable from, IComparable to) : base( from, to )
         {
-            // Issue #21 - https://github.com/nunit/nunit-framework/issues/21
-            // from must be less than or equal to to
-            if (comparer.Compare(from, to) > 0)
-                throw new ArgumentException( "from must be less than to" );
+            this.from = from;
+            this.to = to;
+            CompareFromAndTo();
```

Comment:
You would not believe how much I hate to bring this up.
If we change this constructor from (IComparable, IComparable) to (object, object), we will have to remove this `CompareFromAndTo()` and add it to `Apply` instead. The reason is that if the objects are not comparable, this constructor will throw: [NUnitComparer.cs#L81](https://github.com/JustinRChou/nunit/blob/1f97672f5c0c4ad4093ba48bad68af0523dab828/src/NUnitFramework/framework/Constraints/NUnitComparer.cs#L81)

We can't `CompareFromAndTo()` until we have a comparer, or else .InRange(notComparable, notComparable).Using(customComparer)` will throw at the `.InRange(notComparable, notComparable)` part.

@CharliePoole should this change our minds about providing only (IComparable, IComparable) and (object, object, IComparer)?


That test I wanted awhile back would have poked us in the eye with this.

/cc @CharliePoole @rprouse